h3-quinn/src/lib.rs

```diff
@@ -0,0 +1,286 @@
+//! QUIC Transport implementation with Quinn
+//!
+//! This module implements QUIC traits with Quinn.
+use std::{
+    collections::BTreeMap,
+    task::{self, Poll},
+};
+
+use futures::{ready, FutureExt, StreamExt};
+
+use bytes::{Buf, Bytes};
+use quinn::{
+    generic::{
+        Connection, IncomingBiStreams, IncomingUniStreams, OpenBi, OpenUni, RecvStream, SendStream,
+    },
+    ConnectionError, ReadError, VarInt, WriteError,
+};
+use quinn_proto::crypto::Session;
+
+use h3::quic;
+
+struct QuinnConnection<S: Session> {
```

Comment:
shouldn't it be public, so that it can then be passed to `h3::server::Connection::handshake`?

Reply:
Yes that should! Plus a `new(quinn::Connection)` function should be handy here. Thanks!

Reply:
also NIT: since this is already under the quinn-h3 name could be just Connection and then let the importer rename if conflicting

Reply:
Do you mean `QuinnConnection` should be named `Connection` ?

Reply:
yeah, and then if/when naming conflicts occur, you can import with `use quinn_h3::Connection as QuinnConnection`

Reply:
That's actually a lot better, I thought I would need those prefixes, but no.